keychain-sdk/sign_requests.go

```diff
@@ -23,15 +23,26 @@ type signResponseWriter struct {
 	ctx           context.Context
 	txWriter      *TxWriter
 	signRequestID uint64
+	encryptionKey []byte
 	logger        *slog.Logger
 	onComplete    func()
 }
 
 func (w *signResponseWriter) Fulfil(signature []byte) error {
 	w.logger.Debug("fulfilling sign request", "id", w.signRequestID, "signature", hex.EncodeToString(signature))
+
+	result := signature
+	if w.encryptionKey != nil {
+		var err error
+		result, err = Encrypt(w.encryptionKey, result)
+		if err != nil {
```

Comment:
hmm I think that the correct thing to do for the SDK would be to validate EncyrptionKey before even calling the handler

if it fails validation, it can automatically reject the request

Reply:
it's a good point, I added this in 73d4d53